### PR TITLE
Fed up with notices

### PR DESCRIPTION
**Pull request by hypeJunction**
notices

### CODE REVIEW

File: engine/classes/Elgg/Database/AdminNotices.php [hunk 1]
```diff
@@ -61,27 +65,26 @@ function add($id, $message) {
...
-		
+	public function delete($id = '') {
 		$result = true;
 		
-		$notices = elgg_get_entities([
-			'type' => 'object',
-			'subtype' => 'admin_notice',
+		$notices = $this->find([
 			'metadata_name' => 'admin_notice_id',
 			'metadata_value' => $id,
```
Comment (jeabakker) [inline engine/classes/Elgg/Database/AdminNotices.php]: maybe you could also combine these in a metadata_name_value_pairs? I dislike the separate ones

**Review by jeabakker — APPROVED** [review]

File: engine/classes/Elgg/Database/AdminNotices.php (hunk 1 above)
Comment (hypeJunction) [reply engine/classes/Elgg/Database/AdminNotices.php]: Maybe in some other lifetime

Comment (jdalsem) [reply engine/classes/Elgg/Database/AdminNotices.php]: well this is not what you actually want... plus it is a more complicated query... as it creates an extra join if i am correct

File: engine/classes/Elgg/Database/AdminNotices.php [hunk 2]
```diff
@@ -61,27 +65,26 @@ function add($id, $message) {
 	 *
 	 * @return bool
 	 */
-	function delete($id) {
-		if (!$id) {
-			return false;
-		}
-		
+	public function delete($id = '') {
```
Comment (jdalsem) [inline engine/classes/Elgg/Database/AdminNotices.php]: why no quick return if empty $id?? if you need a delete all just create a new function

File: engine/classes/Elgg/Database/AdminNotices.php (hunk 1 above)
Comment (hypeJunction) [reply engine/classes/Elgg/Database/AdminNotices.php]: QueryBuilder is smart enough to determine the correct query to use based on number of metadata wheres

File: views/default/page/elements/admin_notices.php [hunk 3]
```diff
@@ -4,10 +4,39 @@
...
+
+$notices = elgg_get_admin_notices([
+	'limit' => 5,
+]);
+
+$list = elgg_view_entity_list($notices, [
 	'list_class' => 'elgg-admin-notices',
 ]);
+
+echo elgg_view_module('admin-notices', ' ', $list, [
```
Comment (jdalsem) [inline views/default/page/elements/admin_notices.php]: is the blank space needed?

**Review by jdalsem — CHANGES_REQUESTED** [review]

File: engine/classes/Elgg/Database/AdminNotices.php (hunk 1 above)
Comment (jdalsem) [reply engine/classes/Elgg/Database/AdminNotices.php]: can you show me where??

File: views/default/page/elements/admin_notices.php (hunk 3 above)
Comment (hypeJunction) [reply views/default/page/elements/admin_notices.php]: Yes, module view needs reworking.

File: engine/classes/Elgg/Database/AdminNotices.php (hunk 2 above)
Comment (hypeJunction) [reply engine/classes/Elgg/Database/AdminNotices.php]: Metadata can be empty

Comment (jdalsem) [reply engine/classes/Elgg/Database/AdminNotices.php]: when? and why would you want that usecase in this function? if there is no `id` value than you can not delete by `id`...

Comment (hypeJunction) [reply engine/classes/Elgg/Database/AdminNotices.php]: It's an object, you can change it's metadata to whatever you please. Empty string is valid metadata and valid id.

Comment (jdalsem) [reply engine/classes/Elgg/Database/AdminNotices.php]: i understand that it all is valid, but it is not what the function is for... creation with an empty id is not allowed, so it makes logical sense that the deletion can not happen without it... was there a specific use case why you needed this change?

Comment (hypeJunction) [reply engine/classes/Elgg/Database/AdminNotices.php]: It doesn't make sense to require an attribute and then check if attribute is empty, when we can just allow an empty attribute.

Comment (hypeJunction) [reply engine/classes/Elgg/Database/AdminNotices.php]: Even scruitinizer thinks so https://scrutinizer-ci.com/g/Elgg/Elgg/inspections/58b5ea3b-2cb2-45c5-9991-4f61b65421fd

File: engine/classes/Elgg/Database/AdminNotices.php (hunk 1 above)
Comment (hypeJunction) [reply engine/classes/Elgg/Database/AdminNotices.php]: https://github.com/Elgg/Elgg/blob/master/engine/classes/Elgg/Database/Entities.php#L255

Comment (jdalsem) [reply engine/classes/Elgg/Database/AdminNotices.php]: sweet...

File: engine/classes/Elgg/Database/AdminNotices.php (hunk 2 above)
Comment (jdalsem) [reply engine/classes/Elgg/Database/AdminNotices.php]: could you add a test for `elgg_delete_admin_notice('')`?

Comment (hypeJunction) [reply engine/classes/Elgg/Database/AdminNotices.php]: No, I am done with this PR. Take it or leave it

Comment (hypeJunction) [reply engine/classes/Elgg/Database/AdminNotices.php]: When I ask you to add tests, you do this: https://github.com/Elgg/Elgg/issues/11528

Comment (jdalsem) [reply engine/classes/Elgg/Database/AdminNotices.php]: i thought you were the one that wanted all the tests... i think this is a rare case that gets easily broken, so a test would be good

Comment (jdalsem) [reply engine/classes/Elgg/Database/AdminNotices.php]: > When I ask you to add tests, you do this: #11528

and then it gets merged :)

**Pull request merged**